### PR TITLE
Various colony creation fixes

### PR DESCRIPTION
**Pull request by jakubcolony**
## Description

This PR fixes the following issues with colony creation:
- Setting the newly created token decimals to 0, which was causing the payment action to fail.
- Caching the results of query checking if colony name is already in use, which resulted in 404 after colony create redirection.
- Being unable to activate tokens due to token authority not being set correctly. This had to be fixed on the contracts side so the network image hash was updated.

It also introduces a simpler query to check if colony name is in use as well as refactors `ColonyContext` to remove some of the redundant code.

### CODE REVIEW

File: src/components/v5/common/CreateColonyWizard/validation.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ const COLONY_NAME_REGEX = /^[A-Za-z0-9]\w+$/;
 const TOKEN_SYMBOL_REGEX = /^[A-Za-z0-9]+$/;
 
 const isNameTaken = createYupTestFromQuery({
-  query: GetFullColonyByNameDocument,
+  query: CheckColonyNameExistsDocument,
```
Comment (jakubcolony) [inline src/components/v5/common/CreateColonyWizard/validation.ts:22]: Little point in getting the massive colony object just to see if it exists

**Review by arrenv — APPROVED** [review]
Nice work!! Fixes it for me!

File: src/context/ColonyContext/ColonyContext.tsx [hunk 2]
```diff
@@ -51,31 +46,13 @@ const MSG = defineMessages({
 });
 
 const MIN_SUPPORTED_COLONY_VERSION = 5;
-export const PREV_COLONY_LOCAL_STORAGE_KEY = 'prevColonyName';
```
Comment (chmanie) [inline src/context/ColonyContext/ColonyContext.tsx:54]: Very good, removing this, I approve!

File: src/context/ColonyContext/usePreviousColony.ts
Comment (chmanie) [inline src/context/ColonyContext/usePreviousColony.ts:1]: :+1:

**Review by chmanie — APPROVED** [review]
Really nice work and good clean-up @jakubcolony! I approve!

**Pull request merged**